### PR TITLE
add support of clang compiler && fixed issues found by its static ana…

### PR DESCRIPTION
**Pull request by abudnik**
…lyzer

### CODE REVIEW

**Comment by abudnik** [comment]
BTW: http://www.cplusplus.com/forum/general/186411/

File: library/net.cpp [hunk 1]
```diff
@@ -290,23 +290,18 @@ static bool dnet_recv_nolock(const dnet_connect_state_ptr &state, dnet_addr_sock
 {
 	ssize_t err = recv(socket->s, socket->io_data, socket->io_size, 0);
 	if (err < 0) {
-		err = -EAGAIN;
 		if (errno != EAGAIN && errno != EINTR) {
-			err = -errno;
 			DNET_LOG_ERROR(state->node, "{}: failed to receive data, socket: {}",
```
Comment (shaitan) [inline library/net.cpp]: `DNET_LOG_ERROR` can change `errno`, so it is better to save it before.

Comment (abudnik) [inline library/net.cpp]: done

**Comment by abudnik** [comment]
Analyzer report: http://pastebin.com/zHcZbT1b

**Pull request merged**